FirebaseStorageSwift/Sources/SwiftApiExtension.swift

```diff
@@ -0,0 +1,112 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseStorage
+
+/// getResultCallback generates a closure that returns a Result type from a closure that returns an optional type and Error.
+private func getResultCallback<T>(
+  completion: @escaping (Result<T, Error>) -> Void
+) -> (_: T?, _: Error?) -> Void {
+  return { (metadata: T?, error: Error?) -> Void in
+    guard let metadata = metadata else {
+      guard let error = error else {
+        completion(.failure(NSError(domain: "FirebaseStorageSwift",
+                                    code: -1,
+                                    userInfo: ["Storage Result Generator":
+                                      "InternalError - Return type and Error code both nil"])))
+        return
+      }
+      completion(.failure(error))
+      return
+    }
+    completion(.success(metadata))
+  }
+}
+
+public extension StorageReference {
+  func downloadURL(completion: @escaping (Result<URL, Error>) -> Void) {
```

Comment:
These methods should have comments on par with (or likely copies of) those found in the main class, e.g. https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseStorage/Sources/Public/FIRStorageReference.h#L194.

Feel free to address in a follow-up PR.